app/javascript/packs/entity_autocomplete.js

```diff
@@ -0,0 +1,49 @@
+export default function EntityAutocomplete({config}){
+
+  this.prepare = function(){
+    $(config['input_id']).typeahead(null, {
+      async: true,
+      name: 'entities',
+      source: entitySearch,
+      limit: 8,
+      display: 'name',
+      templates: {
+        empty: $(config['templates']['empty_message']).html(),
+        suggestion: function(data) {
+          return entitySuggestionTemplate().render(data);
+        }
+      }
+    })
+
+    $(config['input_id']).bind('typeahead:select', function(ev, suggestion) {
+      submitEntity(suggestion.id);
+    })
+  };
+
+  const entitySuggestionTemplate = function(){
+    return hogan.compile($(config['templates']['entity_suggestion']).html());
```

Comment:
FYI, one of the things on my list of refactoring is removing hogan FYI!  It's an old library that doesn't get any more updates. There's https://github.com/janl/mustache.js we could replace it with.

Reply:
Happy to substitute this in.

Reply:
OK, I've swapped in mustache.js here and annotated the un-refactored files with a TODO.